--- a/src/main/java/ch/njol/skript/command/ScriptCommand.java
+++ b/src/main/java/ch/njol/skript/command/ScriptCommand.java
@@ -102,6 +_,7 @@
 	@Nullable
 	private final Expression<String> cooldownStorage;
 	final String usage;
+	public String skratches$getUsage() {return usage;}
 
 	private final Trigger trigger;
 
@@ -313,7 +_,7 @@
 			Skript.info("# /" + name + " " + rest);
 		final long startTrigger = System.nanoTime();
 
-		if (!trigger.execute(event))
+		if (!trigger.forceExecute(event))
 			sender.sendMessage(Commands.m_internal_error.toString());
 
 		if (Skript.log(Verbosity.VERY_HIGH))
