--- a/src/main/java/ch/njol/skript/ScriptLoader.java
+++ b/src/main/java/ch/njol/skript/ScriptLoader.java
@@ -811,8 +_,7 @@
 		}
 
 		try {
-			String name = Skript.getInstance().getDataFolder().toPath().toAbsolutePath()
-					.resolve(Skript.SCRIPTSFOLDER).relativize(file.toPath().toAbsolutePath()).toString();
+			String name = file.getCanonicalPath();
 			return loadStructure(Files.newInputStream(file.toPath()), name);
 		} catch (IOException e) {
 			Skript.error("Could not load " + file.getName() + ": " + ExceptionUtils.toString(e));
@@ -892,7 +_,10 @@
 			Structure structure = unloadingStructure.structure();
 
 			parser.setActive(script);
+			try {
 			structure.unload();
+			} catch (Throwable ignored) {
+			}
 		}
 
 		parser.setInactive();
