--- a/src/main/java/ch/njol/skript/lang/SkriptParser.java
+++ b/src/main/java/ch/njol/skript/lang/SkriptParser.java
@@ -280,6 +_,10 @@
 								break;
 						}
 						log.printLog();
+						try {
+							getParser().getCurrentScript().addons.add(JavaPlugin.getProvidingPlugin(info.type()).getName());
+						} catch (Throwable ignored) {
+						}
 						if (doSimplification && element instanceof Simplifiable<?> simplifiable)
 							//noinspection unchecked
 							return (T) simplifiable.simplify();
