--- a/src/main/java/ch/njol/skript/lang/SkriptParser.java
+++ b/src/main/java/ch/njol/skript/lang/SkriptParser.java
@@ -277,6 +_,10 @@
 								break;
 						}
 						log.printLog();
+						try {
+							getParser().getCurrentScript().addons.add(org.bukkit.plugin.java.JavaPlugin.getProvidingPlugin(element.getClass()).getName());
+						} catch (Exception ignored) {
+						}
 						if (doSimplification && element instanceof Simplifiable<?> simplifiable)
 							//noinspection unchecked
 							return (T) simplifiable.simplify();
