--- a/src/main/java/ch/njol/skript/structures/StructAutoReload.java
+++ b/src/main/java/ch/njol/skript/structures/StructAutoReload.java
@@ -116,6 +_,7 @@
 
 	@Override
 	public boolean postLoad() {
+		if (!((me.glicz.skanalyzer.server.AnalyzerServer) Bukkit.getServer()).getAnalyzerConfig().allowAutoReload) return true;
 		task = new Task(Skript.getInstance(), 0, 20 * 2, true) {
 			@Override
 			public void run() {
@@ -146,6 +_,7 @@
 
 	@Override
 	public void unload() {
+		if (task == null) return;
 		task.cancel();
 	}
 
