--- a/src/main/java/ch/njol/skript/ScriptLoader.java
+++ b/src/main/java/ch/njol/skript/ScriptLoader.java
@@ -781,8 +_,11 @@
 		}
 
 		try {
+			/*
 			String name = Skript.getInstance().getDataFolder().toPath().toAbsolutePath()
 					.resolve(Skript.SCRIPTSFOLDER).relativize(file.toPath().toAbsolutePath()).toString();
+			 */
+			String name = file.getCanonicalPath();
 			return loadStructure(Files.newInputStream(file.toPath()), name);
 		} catch (IOException e) {
 			Skript.error("Could not load " + file.getName() + ": " + ExceptionUtils.toString(e));
@@ -848,7 +_,10 @@
 					.forEach(event -> event.onUnload(parser, script));
 
 			for (Structure structure : script.getStructures())
+				try {
 				structure.unload();
+				} catch (Throwable ignored) {
+				}
 		}
 
 		parser.setInactive();
