--- a/src/main/java/ch/njol/skript/lang/SkriptParser.java
+++ b/src/main/java/ch/njol/skript/lang/SkriptParser.java
@@ -259,6 +_,10 @@
 							T element = info.getElementClass().newInstance();
 							if (element.init(parseResult.exprs, patternIndex, getParser().getHasDelayBefore(), parseResult)) {
 								log.printLog();
+								try {
+									getParser().getCurrentScript().addons.add(JavaPlugin.getProvidingPlugin(info.getElementClass()).getName());
+								} catch (Throwable ignored) {
+								}
 								return element;
 							}
 						}
